components/plurals/src/provider.rs

```diff
@@ -198,9 +224,48 @@ impl<'de> serde::Deserialize<'de> for UnvalidatedPluralRange {
     where
         D: serde::Deserializer<'de>,
     {
+        use serde::de::{Error, Visitor};
+
+        struct HumanReadableVisitor;
+
+        impl<'de> Visitor<'de> for HumanReadableVisitor {
+            type Value = UnvalidatedPluralRange;
+
+            fn expecting(&self, formatter: &mut alloc::fmt::Formatter) -> alloc::fmt::Result {
+                write!(
+                    formatter,
+                    "a plural range of the form <PluralCategory>-<PluralCategory>",
+                )
+            }
+
+            fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+            where
+                E: Error,
+            {
+                const VARIANTS: [&str; 6] = [
+                    RawPluralCategory::Other.as_str(),
+                    RawPluralCategory::Zero.as_str(),
+                    RawPluralCategory::One.as_str(),
+                    RawPluralCategory::Two.as_str(),
+                    RawPluralCategory::Few.as_str(),
+                    RawPluralCategory::Many.as_str(),
+                ];
+
+                let (start, end) = v
+                    .split_once('-')
```

Comment:
Thought: We should probably use `"--"` as the separator because it's not out of the question that CLDR could add a BCP-47-like plural form in the future like "other-alt-formal" or something like that.